src/pybind/rados/rados.pyx

```diff
@@ -2548,6 +2549,45 @@ cdef class Ioctx(object):
         if ret < 0:
             raise make_ex(ret, "error flushing")
 
+    @requires(('object_name', str_type), ('cmp_buf', bytes), ('offset', int))
+    def aio_cmpext(self, object_name, cmp_buf, offset=0, oncomplete=None):
+        """
+        Asynchronously compare an on-disk object range with a buffer
+        :param object_name: the name of the object
+        :type object_name: str
+        :param cmp_buf: buffer containing bytes to be compared with object contents
+        :type cmp_buf: bytes
+        :param offset: object byte offset at which to start the comparison
+        :type offset: int
+        :param oncomplete: what to do when the write is safe and complete in memory
+            on all replicas
+        :type oncomplete: completion
+
+        :raises: :class:`TypeError`
+        returns: 0 - on success, negative error code on failure,
+                 (-MAX_ERRNO - dismatch_off) on dismatch
+        """
+        object_name = cstr(object_name, 'object_name')
+
+        cdef:
+            Completion completion
+            char* _object_name = object_name
+            char *_cmp_buf = cmp_buf
+            size_t _cmp_buf_len = len(cmp_buf)
+            uint64_t _offset = offset
+
+        completion = self.__get_completion(oncomplete, None)
+        self.__track_completion(completion)
+
+        with nogil:
+            ret = rados_aio_cmpext(self.io, _object_name, completion.rados_comp,
+                                   _cmp_buf, _cmp_buf_len, _offset)
+
+        if -4096 < ret < 0:
```

Comment:
requires definition in C_API as well which could be used here, instead of hardcoded value as we discussed [here](https://github.com/ceph/ceph/pull/31912#discussion_r357586717)

Reply:
updated, please check.

Reply:
```
int librados::IoCtxImpl:aio_cmpext(const object_t& oid,
                                    AioCompletionImpl *c,
                                    uint64_t off,
                                    bufferlist& cmp_bl)
{
  if (cmp_bl.length() > UINT_MAX/2)
    return -E2BIG;

  Context *onack = new C_aio_Complete(c);

  c->is_read = true;
  c->io = this;

  Objecter::Op *o = objecter->prepare_cmpext_op(
    oid, oloc, off, cmp_bl, snap_seq, 0,
    onack, &c->objver);
  objecter->op_submit(o, &c->tid);

  return 0;
}

```
@ ideepika  As you can see from the code above,  ret< 0 is fine.

Reply:
@ideepika  please check, thanks.